storybookjs/storybook
# feat: split props and slots in vue 3

Issue: #12850 

## What I did

Added non-enumerable sections (containing slots, props etc) to the results of extractArgTypes in Vue 3. 

In Vue 3 the current example extracts all args and binds them to the component. This means that not only props are extracted but also any other values such as slots. The current system also has an issue with slots and props with identical names.

Splitting them this way without affecting the current result usage, so is non-breaking.

## How to test

- All sections should appear as a non-enumerable property of args
- All slots should be in args.slots
- All props should be in args.props

- Does this need a new example in the kitchen sink apps?

Yes, we should recommend using args.slots and args.props.

- Does this need an update to the documentation?

Yes, we should recommend using args.slots and args.props.

If your answer is yes to any of these, please make sure to include it in your PR.

Apologies, it's Easter weekend. I Will try to catch up on docs/kitchen sink early next week but wanted to submit as alternative to #14465
